db/types/alteration.py

```diff
@@ -0,0 +1,209 @@
+from sqlalchemy import text, DDL, MetaData, Table
+from db.types import base, email
+
+BOOLEAN = "boolean"
+EMAIL = email.QUALIFIED_EMAIL
+INTERVAL = "interval"
+NUMERIC = "numeric"
+TEXT = "text"
+VARCHAR = "varchar"
+
+
+def get_supported_alter_column_types(engine):
+    dialect_types = engine.dialect.ischema_names
+    type_map = {
+        # Default Postgres types
+        "boolean": dialect_types.get("boolean"),
+        "interval": dialect_types.get("interval"),
+        "numeric": dialect_types.get("numeric"),
+        "string": dialect_types.get("name"),
+        # Custom Mathesar types
+        "email": dialect_types.get(email.QUALIFIED_EMAIL)
+    }
+    return {k: v for k, v in type_map.items() if v is not None}
+
+
+def alter_column_type(
+        schema, table_name, column_name, target_type_str, engine
+):
+    _preparer = engine.dialect.identifier_preparer
+    supported_types = get_supported_alter_column_types(engine)
+    target_type = supported_types.get(target_type_str.lower())
+    with engine.begin() as conn:
+        metadata = MetaData(bind=engine, schema=schema)
+        table = Table(
+            table_name, metadata, schema=schema, autoload_with=engine
+        )
+        column = table.columns[column_name]
+        prepared_table_name = _preparer.format_table(table)
+        prepared_column_name = _preparer.format_column(column)
+        prepared_type_name = target_type().compile(dialect=engine.dialect)
+        cast_function_name = get_cast_function_name(prepared_type_name)
+        alter_stmt = f"""
+        ALTER TABLE {prepared_table_name}
+          ALTER COLUMN {prepared_column_name}
+          TYPE {prepared_type_name}
+          USING {cast_function_name}({prepared_column_name});
+        """
+        conn.execute(DDL(alter_stmt))
+
+
+def install_all_casts(engine):
+    create_boolean_casts(engine)
+    create_email_casts(engine)
+    create_interval_casts(engine)
+    create_numeric_casts(engine)
+    create_varchar_casts(engine)
+
+
+def create_boolean_casts(engine):
```

Comment:
Could you document what this function (and other similar functions) are doing in comments?

Reply:
Good idea.

Reply:
I made them docstrings to make it visible using the `help` function.  I also considered documenting them in comments in the `PL/pgSQL` functions themselves instead, but I think folks are more likely to be able to (or want to) read the python docstrings than comments in the SQL.  Do you agree?

Reply:
Yes, agreed.